### ISSUE TITLE
Can we use tspOrder2 to order a denovo map?

### ISSUE THREAD

**Issue by xiekunwhy** [pinbo/tspOrder2#1]
Hi,

Can we use tspOrder2 to order a denovo map? For example, all markers produced from stacks (https://catchenlab.life.illinois.edu/stacks/) results.

Best,
Kun

**Comment by pinbo** [comment]
tspOrder2 uses recombination frequencies among markers to order them (genetic distances among markers). If you can get the recombination frequencies (or any other distances) of the markers, then it should be able to order them with the TSP algorithm.

You can read the TSPmap paper to know more (https://biodatamining.biomedcentral.com/articles/10.1186/s13040-017-0158-0), but all it needs is recombination frequencies among markers.

**Comment by xiekunwhy** [comment]
Hi,

I found that you use est.rf() in R/qtl to estimate recombination frequencies, but it may cost huge RAM when marker number in a single LG is large (you can try 10K-50K marker in a single LG). A low-memory cost way to estimate recombination frequencies (joinmap store these into a pwd file) and to prepare Concorde and LKH2 input file is needed.

Best,
Kun

**Comment by pinbo** [comment]
Yes, I suppose so. but, usually, I will not have that many markers if removing colocated markers first. You can use other methods to calculate rf matrix, I just added a function using the rf matrix as input.
